python-checks/src/main/java/org/sonar/python/checks/UselessStatementCheck.java

```diff
@@ -123,9 +125,22 @@ private static void checkNode(SubscriptionContext ctx) {
     if (isWithinIgnoredContext(tree)) {
       return;
     }
+    // Safe cast because the rule only subscribes to expressions
+    if (isAnAirflowException((Expression) tree)) {
+      return;
+    }
     ctx.addIssue(tree, MESSAGE);
   }
 
+  private static boolean isAnAirflowException(Expression expression) {
+    if (isWithinAirflowContext(expression) && expression.type().canBeOrExtend("airflow.models.baseoperator.BaseOperator")) {
```

Comment:
I think we can be lenient here and not actually check whether the expression type is an Airflow operator.
I think focusing the exception on the last statement of the block is already strict enough (I was personally tempted to just keep it disabled in all Airflow context).

Our main goal is not to avoid FNs at all cost, it is to make sure the rule is not noisy when dealing with edge cases/frameworks that may alter the runtime behavior in a way we don't understand.

I also feel we are missing a test where the expression actually isn't a `BaseOperator` despite meeting all other conditions to raise the issue (while coverage is 100%, removing this particular condition doesn't break any test).